src/machine.rs

```diff
@@ -84,9 +84,12 @@ pub struct MemoryExtra {
     /// An allocation ID to report when it is being allocated
     /// (helps for debugging memory leaks).
     tracked_alloc_id: Option<AllocId>,
+
+    /// Place where the `environ` static is stored. Lazily initialized, but then never changes.
+    pub(crate) environ: Option<MPlaceTy<'tcx, Tag>>,
```

Comment:
I just realized we already have the `EnvVars` struct for track extra state for environment variables... @christianpoveda would you mind submitting a follow-up PR to move `environ` over to that struct?

Reply:
sure